repos/spack_repo/builtin/packages/acts/package.py

```diff
@@ -427,6 +433,8 @@ class Acts(CMakePackage, CudaPackage):
     depends_on("mlpack@3.1.1:", when="+mlpack")
     depends_on("nlohmann-json @3.9.1:", when="@0.14: +json")
     depends_on("nlohmann-json @3.10.5:", when="@37: +json")
+    depends_on("torch-scatter", when="+gnn")
+    depends_on("torch-scatter +cuda", when="+cuda")
```

Comment:
We don't want to depend on torch-scatter when just cuda is set but gnn is not.
```suggestion
    depends_on("torch-scatter +cuda +gnn", when="+cuda +gnn")
```

Reply:
Doesn't this need when +torch too then? Or should there be another relationship between the gnn and torch variants?

Reply:
This is independent in principle. You can build `acts +gnn~torch` and you will have a `gnn` plugin that can e.g. run onnx inference, but not torch based inference. Regardless of the choice of inference engine, you will still need `torch-scatter` for parts of the graph building (that happens outside any model inference).

In practice `torch-scatter` will pull in `py-torch` in any case and make sure that all the dependencies are there, but `+torch` is still necessary for acts to enable the necessary cmake option for building the torch based model inference (and ensure `py-torch is also in the build environment).

Reply:
It's more the other way round. Right now `~gnn+torch` is not really any different from `~gnn~torch`, so that seems to indicate that `torch` should only be defined when `+gnn`. Unless there is shortly going to be more `torch` functionality in ACTS that is independent of the GNN plugin.

Reply:
Ah, OK. Sorry, didn't get the question then the first time. Essentially, currently it should be like this, IIUC?

```python
variant("torch", when="+gnn")
```

Reply:
Yes, that's what I meant. I wasn't clear in my first comment and had it inverted myself :-)